src/assets/src/hooks/useWebSocket.ts

```diff
@@ -1,75 +1,101 @@
 import { useState, useEffect } from "react";
-import ReconnectingWebSocket from 'reconnecting-websocket';
+import { WebSocket } from "partysocket";
 
 interface OfficeHoursMessage<T> {
-    type: "init"|"update"|"deleted";
-    content: T;
+  type: "init" | "update" | "deleted";
+  content: T;
 }
 
-export const useWebSocket = <T>(url: string, onUpdate: (content: T) => void, onDelete?: (setError: (React.Dispatch<React.SetStateAction<Error | undefined>>)) => void) => {
-    const [error, setError] = useState(undefined as Error | undefined);
-    const numRetries = 10;
-    useEffect(() => {
-        const ws = new ReconnectingWebSocket(
-            url,
-            undefined,
-            { maxRetries: numRetries },
+export const useWebSocket = <T>(
+  url: string,
+  onUpdate: (content: T) => void,
+  onDelete?: (
+    setError: React.Dispatch<React.SetStateAction<Error | undefined>>
+  ) => void
+) => {
+  const [error, setError] = useState(undefined as Error | undefined);
+  useEffect(() => {
+    const ws = new WebSocket(url, undefined, {
+      maxRetries: 10,
+    });
+    ws.onmessage = (e: MessageEvent) => {
+      const m = JSON.parse(e.data) as OfficeHoursMessage<T>;
+      console.log(m);
+      switch (m.type) {
+        case "init":
```

Comment:
These init and update perform exactly the same action. They can be combined to reduce redundancy

```
  case "init":
  case "update":
    onUpdate(m.content as T);
    break;
```